src/operations/mutations/addCollections.tsx

```diff
@@ -1,16 +1,30 @@
 import { gql, useMutation } from '@apollo/client'
-import type { Collection, CollectionRecords } from 'types'
+import { WIDGET_TYPES } from 'constants/index'
+import type {
+  BookmarkRecordInput,
+  Collection,
+  CollectionRecordInput,
+} from 'types'
 
 interface AddCollectionsResponse {
   collections: Collection[]
 }
 
 interface AddCollectionsInput {
-  collections: CollectionRecords
+  collections: CollectionRecordInput[]
+}
+
+export const addCollectionsInput = (c: CollectionRecordInput[]) => {
```

Comment:
Ok, my understanding of this function is that it's taking Collection data from the CMS (`CollectionRecords`), and mapping it to the input expected by the `addCollections` operation to be inserted into Mongo (`CollectionRecordInput[]`), right? I was a little bit confused & I think this is a good place where we could use more explicit types to make the purpose of the function a little clearer.

For example, if these are the type definitions:
```
export type BookmarkRecord = {
  id: string
  url: string
  label?: string
  description?: string
}

export type CollectionRecord = {
  id: string
  title: string
  bookmarks: BookmarkRecords
}

export type CollectionRecordInput = Pick<
  CollectionRecord,
  'id' | 'title' | 'bookmarks'
>

export type BookmarkRecordInput = Pick<BookmarkRecord, 'id' | 'url' | 'label'>
```

the function can be typed like so:
```
export const addCollectionsInput = (
  c: CollectionRecords
): CollectionRecordInput[] => {
  return c.map(({ id, title, bookmarks }) => {
    return {
      id,
      title,
      bookmarks: bookmarks.map(({ id, url, label }): BookmarkRecordInput => {
        return { id, url, label }
      }),
    }
  })
}

``` 

Also I don't think this function needs to add the `type` property, because that is explicitly added by the `Collection.addMany` model.